src/Razor/test/Microsoft.VisualStudio.Editor.Razor.Test/TelemetryReporterTests.cs

```diff
@@ -322,4 +322,108 @@ public void TrackLspRequest()
                 Assert.Equal(correlationId, correlationProperty.Value);
             });
     }
+
+    [Theory]
+    [InlineData(typeof(RemoteInvocationException), 3)]
+    [InlineData(typeof(OperationCanceledException), 3)]
+    [InlineData(typeof(AggregateException), 15000)]
+    public void ReportFault_MixedTypeNestedExceptions_NoStackOverflowException(Type exceptionType, int depth)
+    {
+        // Arrange
+        var reporter = new TestTelemetryReporter(new RazorLoggerFactory([]));
+        var exception = new RemoteInvocationException("Test", 0,
+                innerException: CreateNestedException(typeof(AggregateException), 3,
+                    innerMostException: CreateNestedException(exceptionType, depth,
+                        innerMostException: new ApplicationException("expectedText")
+                    )
+                )
+            );
+
+        // Act
+        reporter.ReportFault(exception, exception.Message);
+
+        // Assert
+        Assert.Collection(reporter.Events,
+            e1 =>
+            {
+                Assert.Equal("dotnet/razor/fault", e1.Name);
+            });
+    }
+
+    [Theory]
+    [InlineData(typeof(RemoteInvocationException), 3)]
+    [InlineData(typeof(OperationCanceledException), 3)]
+    [InlineData(typeof(AggregateException), 15000)]
+    public void ReportFault_NestedException_NoStackOverflowException(Type exceptionType, int depth)
```

Comment:
FYI, `ReportFault_NestedException_NoStackOverflowException` would throw stackoverflow exception for nested types `RemoteInvocationException` and `OperationCanceledException` when depth is larger